drivers/scheduler/k8s/k8s.go

```diff
@@ -628,11 +628,11 @@ func (k *k8s) substituteNamespaceInVolumes(volumes []v1.Volume, ns string) []v1.
 	return updatedVolumes
 }
 
-func (k *k8s) WaitForRunning(ctx *scheduler.Context) error {
+func (k *k8s) WaitForRunning(ctx *scheduler.Context, timeout, retryInterval time.Duration) error {
 	k8sOps := k8s_ops.Instance()
 	for _, spec := range ctx.App.SpecList {
 		if obj, ok := spec.(*apps_api.Deployment); ok {
-			if err := k8sOps.ValidateDeployment(obj); err != nil {
+			if err := k8sOps.ValidateDeployment(obj, timeout, retryInterval); err != nil {
```

Comment:
need to use timeout instead of `statefulSetValidateTimeout` below too. All timeouts in `WaitForRunning` should be proportional to the given `timeout`.